thunder/__init__.py

```diff
@@ -760,18 +765,19 @@ def fn_(*args, **kwargs) -> Any:
 
         if cache_entry.backward_fn:
             # Run the compiled forward function
-            data_for_autograd, (saved_tensors, saved_other) = result
+            result, (saved_tensors, saved_other) = result
 
             # Connect produced tensors with PyTorch's autograd graph
             ThunderFunction.apply(
                 cache_entry.return_none_instead_of_grads,
                 cache_entry.backward_fn,
                 saved_tensors,
                 saved_other,
-                data_for_autograd["flat_output"],
-                *data_for_autograd["flat_args"],
+                result["flat_output"],
+                *result["flat_args"],
             )
-            result = data_for_autograd["output"]
+
+        result = result["output"]
```

Comment:
Instead of this change, could we please add

```
if backward_trc is None:
      computation_trace = unwrap_return_value(computation_trace)
```

here:

https://github.com/Lightning-AI/lightning-thunder/blob/3770bf02240cdb6ec86a6d5224807bc6ad39d2a9/thunder/__init__.py#L670-L671

?
That would leave the inference cases alone.